lib/ansible/plugins/strategy/__init__.py

```diff
@@ -898,7 +892,10 @@ def _evaluate_conditional(h):
                         iterator._host_states[host.name].run_state = iterator.ITERATING_COMPLETE
                 msg = "ending play"
         elif meta_action == 'reset_connection':
-            connection = connection_loader.get(play_context.connection, play_context, os.devnull)
+            all_vars = self._variable_manager.get_vars(play=iterator._play, host=target_host, task=task)
+            templar = Templar(loader=self._loader, variables=all_vars)
+            updated_play_context = self._play_context.set_task_and_variable_override(task=task, variables=all_vars, templar=templar)
```

Comment:
Why is self._play_context updated instead of the passed in play_context?

(where does self._play_context come from? I don't see it set/defined anywhere)

Reply:
It gets further if you change  'self._play_context' to just play_context, but still fails.